src/core/providers/ogr/qgsogrprovidermetadata.cpp

```diff
@@ -1071,6 +1075,118 @@ bool QgsOgrProviderMetadata::saveLayerMetadata( const QString &uri, const QgsLay
   throw QgsNotSupportedException( QObject::tr( "Storing metadata for the specified uri is not supported" ) );
 }
 
+QList<QgsLayerMetadataProviderResult> QgsOgrProviderMetadata::searchLayerMetadata( const QString &uri, const QString &searchString, const QgsRectangle &geographicExtent, QgsFeedback *feedback )
+{
+
+  QList<QgsLayerMetadataProviderResult> results;
+  QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "ogr" ) ) };
+  Q_ASSERT( md );
+  std::unique_ptr<QgsAbstractDatabaseProviderConnection> conn;
+  conn.reset( static_cast<QgsAbstractDatabaseProviderConnection *>( md->createConnection( uri, {} ) ) );
+  if ( conn && ( ! feedback || ! feedback->isCanceled() ) )
+  {
+    try
+    {
+      const QString searchQuery { QStringLiteral( R"SQL(
```

Comment:
There's no gaurantee that QgsOgrProviderMetadata::createConnection gives you back a connection to a gpkg -- it may be QgsOgrProviderConnection instead. Should this API logic be added to QgsAbstractProviderConnection, and then this logic moved to QgsGeoPackageProviderConnection?

Reply:
This was my first idea too, I put it in the metadata because the styles API was already there and the concept of storing md and styles (and projects) in the DB is rather similar.